### DISCUSSION TITLE
Enable adb root

### DISCUSSION THREAD

**Discussion by arjunair86** [LSPosed/MagiskOnWSALocal#611] [Q&A]
adb root results in 

`
adbd cannot run as root in production builds
`

I figured it would require a insecure version of adbd to be installed and patched into WSA.  I also realised that it can be done through magisk module and found something which does it along those lines: https://github.com/tiann/adb_root

But unfortunately above method does not work, due to multiple reasons:
- libs compiled are in arm64
- adbd is in apex file format in WSA 2305

Can someone guide on how we can achieve adb root in WSA 2305?

**Comment by fromhgbwithluv** [comment]
Just build WSA with KernelSu then connect to ADB and type `su`

**Reply by arjunair86** [reply]
Just tried, but building WSA with KernelSU does not seem to work for WSA 2305. There is no su binary I guess. 
Output
`
/system/bin/sh: su: inaccessible or not found
`

**Comment by Almighty188** [comment]
Are you just trying to get to the rooted shell prompt?  or do you need the root for something else?  If for the former:
Open command prompt in Windows known as terminal
adb shell 
which will show
redfin:/ $ su
redfin:/ #

**Reply by arjunair86** [reply]
I wanted adb daemon to run as root (assuming that's what adb root does) to mount system partition as rw. 
Was able to do it via KernelSU and [Magic OverlayFS](https://github.com/HuskyDG/magic_overlayfs) module. Though realised later that I wanted `/apex` directory and all subdirs under it also mounted as rw, which magic_overlayfs does not achieve.

**Comment by arjunair86** [comment]
Closing the discussion. To mount system partitions as rw, install WSA with KernelSU and follow https://github.com/HuskyDG/magic_overlayfs
